### PR TITLE
WIP - Replace deprecated UA

### PR DESCRIPTION
**Pull request by confususs**
### Issue number

#13 

### Moving to the new API

[x] - registering
[x] - subscribing
[x] - placing an outgoing call (+ accepted/rejected/terminated promises)
[x] - rebuilding SDH on connectivity failures
[x] - hold / unhold 
[x] - getting an incoming call
[x] - blind transfer
[x] - attended transfer
[x] - dtmf
[x] - uaOptions should include defaultOptions
~[ ] - look for asterisk specific header in terminated callback~ skipping this for now
[x] - update remote identity on reinvite
[x] - check if autoanswer still works
[x] - use delegate for subscribing
[x] - fix call quality in demo

### CODE REVIEW

File: src/session.ts [hunk 1]
```diff
@@ -290,8 +280,9 @@ export class SessionImpl extends EventEmitter implements ISession {
 
   get autoAnswer(): boolean {
     const callInfo = this.session.request.headers['Call-Info'];
-    if (callInfo && callInfo[0]) {
-      return callInfo[0].raw.includes('answer-after=0');
+    if (callInfo && callInfo[0] && callInfo[0]) {
```
Comment (c0deaddict) [inline src/session.ts]: double check for `callInfo[0]`

File: src/session.ts [hunk 2]
```diff
@@ -526,60 +512,36 @@ export class SessionImpl extends EventEmitter implements ISession {
...
+            resolve(true);
+          },
+          // Refer can be rejected with the following responses:
+          // - 503: Service Unavailable (i.e. server can't handle one-legged transfers)
+          // - 603: Declined
+          onReject: () => {
+            log.info('Transferred session is rejected!', this.constructor.name);
+            resolve(false);
+          },
+          onNotify: () => ({}) // To
```
Comment (c0deaddict) [inline src/session.ts]: To.. Do? :)

File: src/transport.ts [hunk 3]
```diff
@@ -10,32 +18,53 @@ import { log } from './logger';
...
+   * cause it to be blocked. To make sure that UA gets to the proper internal
+   * state so that it is ready to 'switch over' to the new network interface
+   * with a new websocket, we call the function that normally causes the
+   * disconnectPromise to be resolved after a timeout.
+   */
+  protected disconnectPromise(options: any = {}): Promise<any> {
+    return pTimeout(super.disconnectPromise(), 1000, () => {
+      log.debug('Fake-closing the the socket by ourselves.', this.constructor.name);
+      (this as any).onClose({ code: 'fake', reason: 'Artificial timeout' });
+    }).then(() => ({ overrideEvent: true }));
```
Comment (c0deaddict) [inline src/transport.ts]: Document what the overrideEvent return value is for.

File: src/session.ts (hunk 2 above)
Comment (confususs) [reply src/session.ts]: To make sure the requestDelegate type is complete xD

**Pull request merged**